### DISCUSSION TITLE
Help with error in Github pages: jekyll-imagemagick, mkdir: no such file or directory @ dir_s_mkdir - _site

### DISCUSSION THREAD

**Discussion by metalescab** [george-gca/multi-language-al-folio#30] [Q&A]
Hi dear all,

I'm running the theme for my site on GitHub pages.
It was working nicely, but suddenly I got the next error:

``
/home/runner/work/metalescab.github.io/metalescab.github.io/vendor/bundle/ruby/3.2.0/gems/jekyll-imagemagick-1.4.0/lib/jekyll-imagemagick/generator.rb:35:in `mkdir': No such file or directory @ dir_s_mkdir - /home/runner/work/metalescab.github.io/metalescab.github.io/_site/en (Errno::ENOENT)
``

However, I didn't get the same error running locally.

Thank you in advance for any help or advice.
Julio

**Comment by george-gca** [comment]
This is really weird. How are you running locally? Also can you pinpoint when exactly this started happening? The specific commit?

**Reply by metalescab** [reply]
I'm sorry if "running locally" is not the correct term. I mean, on my computer, executing `bundle exec jekyll serve --trace`. Then, when I committed, uploaded, and ran the workflow on GitHub I got that error.

The specific commit when I got that error was Commit https://github.com/metalescab/metalescab.github.io/commit/2326b80c77b93c05c0fa1d2759777c2433731583

**Reply by george-gca** [reply]
I asked how because you could run locally with or without docker, on windows, linux, or mac, so there are a lot of variables here.

Just checked your site and everything seems fine. In this case, if you manage to figure how to fix yourself, it would be useful to share the answer here so others might benefit from it.

**Comment by george-gca** [comment]
Updating to the latest version of the repo might fix this for you. In https://github.com/george-gca/multi-language-al-folio/commit/e46a7941b216c68493158fe412467c1a11fb8b54 I updated all dependencies versions to fix https://github.com/alshedivat/al-folio/security/dependabot/4.

**Reply by metalescab** [reply]
I modified the `gemfile.lock` with your changes and the error persists on GitHub pages. Furthermore, now I have the same running on my computer :(, which I didn't have before.

**Comment by DecAngel** [comment]
I met the same error as @metalescab when changing the default language (in my case changing to zh-cn). I fixed the error by manually creating _site/en-us and _site/zh-cn, filling them with .gitkeep (to make git upload empty directories), removing _site in .gitignore and pushing to main branch. The error message was not useful so I still don't know the reason of the error.

- File Structure
![image](https://github.com/user-attachments/assets/3aef668f-4cb0-4d2f-83f1-ac11d471dbd1)
- Before
![image](https://github.com/user-attachments/assets/33ad2ac7-163b-4771-ac58-7600c3eb7ed5)
- After
![image](https://github.com/user-attachments/assets/ad3349a2-43c4-4821-b061-0d173aab5b3e)

**Reply by george-gca** [reply]
Are you still getting this error @metalescab?
